### PR TITLE
ec/asm/x25519-x86_64.pl: fix base 2^64 addition and final reduction.

### PR DESCRIPTION
**Pull request by dot-asm**
Base 2^64 addition and final reduction failed to treat partially reduced
values correctly.

- [x] tests are added or updated

### CODE REVIEW

**Comment by dot-asm** [comment]
"Thanks to Wycheproof Project for vectors and Paul Kehrer for report"

**Comment by dot-asm** [comment]
Re-mastered first commit to cover even subtraction, removed "ready" label, because re-approval is needed.

**Comment by dot-asm** [comment]
For reference, follow-up commit for curve25519.c does not change any code, only order of #if sections. Originally base 2^64 code was appearing in the middle of base 2^51, which could be hard to follow.

**Comment by dot-asm** [comment]
Re-approval pending.

**Comment by richsalz** [comment]
reconfirm
